builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Security Considerations
+description: A description of the main differences that Ethereum developers need to understand in terms of security considerations when developing on Moonbeam.
+---
+
+# Security Considerations
+
+![Moonbeam v Ethereum - Security Banner](/images/builders/get-started/eth-compare/security-banner.png)
+
+## Introduction {: #introduction }
+
+When developing smart contracts on Moonbeam, there are some security considerations to be aware of that do not apply when developing on Ethereum. Moonbeam has several [precompiled contracts](/builders/pallets-precompiles/precompiles/){target=_blank}, which are Solidity interfaces that enable developers to access Substrate-based functionality through the Ethereum API, but circumventing the EVM. Although the precompiled contracts are designed to improve the developer experience, there can be some unintended consequences that must be considered.
+
+This guide will outline and provide examples of some security considerations to be cognizant of when developing on Moonbeam.
+
+## Arbitrary Code Execution {: #arbitrary-code-execution }
+
+Smart contracts can allow arbitrary code execution through low-level message passing to other smart contracts via the [`call()` function](https://solidity-by-example.org/call/){target=_blank}, which is an available method for the [address data type in Solidity](https://docs.soliditylang.org/en/latest/types.html#address){target=_blank}.
+
+The `call()` function accepts ABI encoded call data that is then passed to the target contract. To get the encoded call data, you can use any of the [ABI encoding functions outlined in the Solidity docs](https://docs.soliditylang.org/en/latest/units-and-global-variables.html#abi-encoding-and-decoding-functions){target=_blank}, including `abi.encodeWithSelector` which will be used in the examples in this guide. 
+
+As previously mentioned, one major concern of allowing low-level calls to arbitrarily execute code on Moonbeam is that Moonbeam has precompile contracts that can be called, which can be used to get around some protections that are typically available on Ethereum. To safely use arbitrary code execution on Moonbeam, you should consider the following, which **only applies to contracts that allow arbitrary code execution**:
```

Comment:
To illustrate my previous point:

`arb code execution` here is a bit confusing. From this context (this paragraph), it could mean:

* code executed from a smart contract in general -- from the EVM's perspective, this is arbitrary (other than following the normal rules)
* `allowing low-level calls to arbitrarily execute code on Moonbeam` is probably nothing different than other cases, but is a lot to swallow
* code that is executed as the target of some `call` (or similar) -- I think this is mainly what this article is concerned with

I think these are really all referring to the same thing, so maybe using (1) consistent phrasing and (2) a short intro giving context to "arb code exec" and esp. that we are referring to it as a result of some `*call` would help a lot.